src/Console/Command/StageCommand.php

```diff
@@ -3,21 +3,72 @@
 namespace PhpTuf\ComposerStager\Console\Command;
 
 use PhpTuf\ComposerStager\Console\Misc\ExitCode;
+use PhpTuf\ComposerStager\Domain\Stager;
 use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 
 class StageCommand extends Command
 {
     protected static $defaultName = 'stage';
 
+    /**
+     * @var \PhpTuf\ComposerStager\Domain\Stager
+     */
+    private $stager;
+
+    public function __construct(Stager $stager)
+    {
+        parent::__construct(static::$defaultName);
+        $this->stager = $stager;
+    }
+
+    /**
+     * @throws \Symfony\Component\Console\Exception\InvalidArgumentException
+     */
     protected function configure(): void
     {
-        $this->setDescription('Stages a Composer command');
+        $this
+            ->setDescription('Stages a Composer command')
+            ->addArgument(
+                'composer-command',
+                InputArgument::IS_ARRAY | InputArgument::REQUIRED,
+                'The raw Composer command to stage. This MUST be preceded by a double-hyphen (" -- ") to prevent confusion of command options. See "Usage"'
+            )
+            ->addUsage('[options] -- <composer-command>...')
+            ->addUsage('-- update --with-all-dependencies')
+            ->addUsage('-- require lorem/ipsum:"^1 || ^2"')
+            ->addUsage('-- --help')
+            ->addUsage('--ansi -- help --ansi')
```

Comment:
This example seems wrong; it's not prefixed with ` -- ` and it repeats `--ansi` twice.

Reply:
That's actually the entire point of that example. Anything before the `--` is interpreted as part of the `composer-stage` command. Anything after the `--` is treated as an argument to send (stage) to Composer. Thus I wanted an example that highlights that--and I felt like passing the same option to each made clearest that it's doing two different things. Hopefully the explicit documentation in the command help will compensate for the (admittedly) uncommon idiom.

Reply:
I think I would probably remove the example, to be honest; when trying to come to grips with the command, I think it will throw peoples' brains for a loop. This "uncommon idiom" (a phrase I really like, by the way) is probably something that should be covered explicitly in online documentation, rather than this more lightweight built-in documentation. Just my $0.02; leave it in if you feel strongly about it :)

Reply:
Okay. I'll think about that when I get to implementing the feedback here.

Reply:
Removed. ¯\_(ツ)_/¯